app/shared/components/design-system/all-components/icon-button/IconButton.tsx

```diff
@@ -0,0 +1,65 @@
+import {
+  CircularProgress,
+  IconButtonProps as MuiIconButtonProps,
+  IconButton as MuiIconButton,
+} from '@mui/material';
+import { FC } from 'react';
+import { IconButtonStyles } from './IconButton.styles';
+import {
+  IconButtonColorVariant,
+  IconButtonVariant,
+} from '~/types/enums/common.enums';
+
+interface IconButtonProps extends Omit<MuiIconButtonProps, 'type'> {
+  variant?: IconButtonColorVariant;
+  outlined?: boolean;
+  size?: 'small' | 'medium' | 'large';
+  loading?: boolean;
+  disabled?: boolean;
+  toggleAble?: boolean;
+  isToggled?: boolean;
+  type?: IconButtonVariant;
+  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
+  children?: React.ReactNode;
+}
+
+export const IconButton: FC<IconButtonProps> = ({
```

Comment:
It's better to use `React.FC` to indicate that it's react type and not some custom.

```suggestion
export const IconButton: React.FC<IconButtonProps> = ({
```

Reply:
Fixed